io/net/virt-net/Network_virtualization_client_initialized_failover.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM
+# Author: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>
+
+import time
+import netifaces
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils.process import CmdError
+
+
+class NetworkVirtualizationFailoverTest(Test):
+
+    """
+    This test performs failover for a Network virtualized device
+    """
+
+    def setUp(self):
+        """
+        Install necessary packages and identify the network
+        virtualized device.
+        """
+        smm = SoftwareManager()
+        for pkg in ["net-tools"]:
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.cancel("%s package is need to test" % pkg)
+        interfaces = netifaces.interfaces()
+        self.interface = self.params.get('interface')
+        if self.interface not in interfaces:
+            self.cancel("%s interface is not available" % self.interface)
+        self.device = process.system_output("ls -l /sys/class/net/ | \
+                                             grep %s | cut -d '/' -f \
+                                             5" % self.interface,
+                                            shell=True).strip()
+        self.count = int(self.params.get('count', default="1"))
+        self.peer_ip = self.params.get('peer_ip', default=None)
+
+    def test(self):
+        '''
+        Performs failover for Network virtualized device
+        '''
+        try:
+            for val in range(self.count):
+                self.log.info("Performing Client initiated\
+                              failover - Attempt %s" % int(val+1))
+                process.run('echo 1 > /sys/devices/vio/%s/failover' %
+                            self.device, shell=True, sudo=True)
+                time.sleep(10)
```

Comment:
Is it always 10s ?

Reply:
Well, it.s less than 10 but definitely above 5. Chose 10 seconds just so it looks visually pleasing.

Reply:
Is there any way to find if failover is complete ?